testdata/dnn/download_models.py

```diff
@@ -85,8 +86,25 @@ def get(self):
         candidate_verify = self.verify()
         if not candidate_verify:
             if os.path.exists(self.filename):
-                print('  deleting invalid file')
-                os.remove(self.filename)
+                try:
+                    if hasattr(self, 'sha_actual') and self.sha_actual:
+                        rename_target = self.filename + '.' + str(self.sha_actual)
```

Comment:
I suggest to have a single invalid name. Not to have always changing invalid names. Difficult to manage cleanup, ever increase drive usage, etc. 👎 Simpler code too.

It is easily possible for a download's `sha_actual` to change on every attempt. The results in unbounded number of invalid files. This is a problem I suggest we don't create. (Similar to log files that don't manage themselves.) If some developer out there needs per-download-attempt-backup-files, they can rename it themselves.

I do not think there is any useful mainstream scenario that needs per-download-attempt-backup-files.
A single xxxx.invalid for the most recent attempt is all that is needed and for it to be replaced with a new attempt.